### PR TITLE
[BK-4133] Fix conversion of dict type

### PR DESCRIPTION
**Pull request by yael-stratoscale**
@Stratoscale/mgmt

### CODE REVIEW

**Comment by shimshon-stratoscale** [comment]
looks ok to me, but @eyal-stratoscale should approve.
You have a failing unit test though

File: hammock/converters.py [hunk 1]
```diff
@@ -28,9 +29,15 @@ def to_list(value):
...
 
 def to_dict(value):
+    if value is None:
+        return {}
+    dict_value = value
     if isinstance(value, six.string_types):
-        value = json.loads(value)
-    return value
+        dict_value = json.loads(value)
+    if not isinstance(dict_value, dict):
```
Comment (adir-stratoscale) [inline hammock/converters.py]: i think that `json.loads` may return a a list (=valid json).

**Comment by eyal-stratoscale** [comment]
Looks great!
Shipit!

**Comment by eyal-stratoscale** [comment]
@shimshon-stratoscale , this test started to be flaky since the read=0 argument we added in the client code.

File: hammock/converters.py [hunk 2]
```diff
@@ -28,9 +29,15 @@ def to_list(value):
...
+        raise exceptions.BadRequest(
+            'Got {} ({}) instead of type dict'.format(value, type(value)))
```
Comment (adir-stratoscale) [inline hammock/converters.py]: no need to raise BadRequest, just raise ValueError - the caller will translate the exception to BadRequest.

Comment (adir-stratoscale) [inline hammock/converters.py]: this where the translation happens:
https://github.com/Stratoscale/hammock/blob/master/hammock/types/func_spec.py#L60

**Pull request merged**